app/models/compute_resource.rb

```diff
@@ -439,10 +439,26 @@ def resolve_automatic_firmware(firmware, firmware_type)
   # @param firmware_type [String] The firmware type based on the provided PXE Loader.
   # @return [Hash] A hash containing the processed firmware attributes.
   def process_firmware_attributes(firmware, firmware_type)
+    attrs = {}
+    # initial_firmware is the firmware type assigned either by default or through the compute profile
+    initial_firmware = firmware if firmware.present?
     firmware = resolve_automatic_firmware(firmware, firmware_type)
 
-    attrs = generate_secure_boot_settings(firmware)
-    attrs[:firmware] = normalize_firmware_type(firmware)
+    attrs.merge!(generate_secure_boot_settings(firmware))
```

Comment:
If `attrs` was empty before then why don't you use
```suggestion
    attrs = generate_secure_boot_settings(firmware)
```

Or if you you want a copy because you're modifying it later:

```suggestion
    attrs = generate_secure_boot_settings(firmware).dup
```

But I don't think that' needed. In other words: why did you modify old line 444?

Reply:
Thanks @ekohl
absolutely correct. It was a left ofter from previous fix.